nima/http2/webserver/src/main/java/io/helidon/nima/http2/webserver/Http2Stream.java

```diff
@@ -384,8 +397,15 @@ private void handle() {
                                                                    streamId,
                                                                    this::readEntityFromPipeline);
             Http2ServerResponse response = new Http2ServerResponse(ctx, request, writer, streamId, flowControl.outbound());
+            semaphoreAcquired = requestSemaphore.tryAcquire();
             try {
-                routing.route(ctx, request, response);
+                if (semaphoreAcquired) {
+                    routing.route(ctx, request, response);
+                } else {
+                    ctx.log(LOGGER, TRACE, "Too many concurrent requests, rejecting request.");
+                    response.status(Http.Status.SERVICE_UNAVAILABLE_503)
+                            .send("Too Many Concurrent Requests");
+                }
```

Comment:
We should have test coverage for this too

Reply:
There is test coverage for all three, 503 in io.helidon.tests.integration.webserver.resourcelimit.MaxConcurrentRequestsTest

Reply:
ah, I see, for HTTP/2

Reply:
Fixed, added HTTP/2 support for testing, added test.